src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+}
+
+static const item get_top_item_at_point( const tripoint &point,
+        const units::volume &min_visible_volume )
+{
+    map_stack items = g->m.i_at( point );
+    // iterate from topmost item down to ground
+    for( const item &it : items ) {
+        if( it.volume() > min_visible_volume ) {
+            // return top (or first big enough) item to the list
+            return it;
+        }
+    }
+    return item();
+}
+
+static std::string colorized_ter_name_flags_at( const tripoint &point,
+        const std::vector<std::string> &flags, const std::vector<std::string> &ter_whitelist )
+{
+    const ter_id ter = g->m.ter( point );
+    std::string name = colorize( ter->name(), ter->color() );
+    const std::string &graffiti_message = g->m.graffiti_at( point );
+    const std::string trap_name = colorized_trap_name_at( point );
+
+    if( !graffiti_message.empty() ) {
+        name += _( " with graffiti \"" ) + graffiti_message + "\"";
+        return name;
+    }
+    if( ter_whitelist.empty() && flags.empty() ) {
+        return name;
+    }
+    if( !ter->open.is_null() || ( ter->examine != iexamine::none &&
+                                  ter->examine != iexamine::fungus &&
+                                  ter->examine != iexamine::water_source &&
+                                  ter->examine != iexamine::dirtmound ) ) {
+        return name;
+    }
+    for( const std::string &ter_good : ter_whitelist ) {
+        if( ter->id.c_str() == ter_good ) {
+            return name;
+        }
+    }
+    for( const std::string &flag : flags ) {
+        if( ter->has_flag( flag ) ) {
+            return name;
+        }
+    }
+
+    return std::string();
+}
+
+static std::string colorized_feature_description_at( const tripoint center_point, bool &item_found,
+        const units::volume &min_visible_volume )
+{
+    item_found = false;
+    const furn_id furn = g->m.furn( center_point );
+    if( furn != f_null && furn.is_valid() ) {
+        std::string furn_str = "<color_yellow>" + furn->name() + "</color>";
+        std::string sign_message = g->m.get_signage( center_point );
+        if( !sign_message.empty() ) {
+            furn_str += _( " with message \"" ) + sign_message + "\"";
+        }
+        if( !furn->has_flag( "CONTAINER" ) && !furn->has_flag( "SEALED" ) ) {
+            const item item = get_top_item_at_point( center_point, min_visible_volume );
+            if( !item.is_null() ) {
+                furn_str += _( " with " ) + colorized_item_name( item );
+                item_found = true;
+            }
+        }
+        return furn_str;
+    }
+    return std::string();
+}
+
+static std::string get_vehicle_hash( const vehicle &veh )
+{
+    const tripoint veh_coord = veh.global_pos3();
+    return string_format( "%s %i %i %i", veh.disp_name(), veh_coord.x, veh_coord.y, veh_coord.z );
+}
+
+static const auto format_object_pair = []( const std::pair<std::string, int> &pair,
```

Comment:
Why is this a lambda (in the global scope) and not just a free function? Same with the next one.

Reply:
Somehow i thought `enumerate_as_string()` takes only a lambda as argument, but function works too, as you pointed out